web/frontend/src/components/do-command/get-client-by-type.ts

```diff
@@ -0,0 +1,71 @@
+import type { RobotClient } from '@viamrobotics/sdk';
+
+const DO_COMMAND_CLIENT_TYPES = [
+  'arm',
+  'base',
+  'board',
+  'encoder',
+  'gantry',
+  'generic',
+  'gripper',
+  'input_controller',
+  'motion',
+  'motor',
+  'movement_sensor',
+  'navigation',
+  'power_sensor',
+  'sensors',
+  'servo',
+  'slam',
+  'vision',
+] as const;
+
+type DoCommandClientType = (typeof DO_COMMAND_CLIENT_TYPES)[number];
+
+type DoCommandServiceType =
+  | 'armService'
+  | 'baseService'
+  | 'boardService'
+  | 'encoderService'
+  | 'gantryService'
+  | 'genericService'
+  | 'gripperService'
+  | 'inputControllerService'
+  | 'motionService'
+  | 'motorService'
+  | 'movementSensorService'
+  | 'navigationService'
+  | 'powerSensorService'
+  | 'sensorsService'
+  | 'servoService'
+  | 'slamService'
+  | 'visionService';
+
+const CLIENT_TYPES: Record<DoCommandClientType, DoCommandServiceType> = {
```

Comment:
This file would be much smaller if this mapping object was defined as `as const` and the `DoCommandClientType` and `DoCommandServiceType` were derived from the object

```ts
const CLIENT_TYPES = { /* ... */ } as const
type DoCommandClientType = keyof typeof CLIENT_TYPES
type DoCommandServiceType = (typeof CLIENT_TYPES)[DoCommandClientType]
```

That being said, since these types aren't exported, what's the point of repeating ourselves here instead of simply defining the map?